server/images.go

```diff
@@ -48,16 +53,43 @@ type Model struct {
 	ParentModel    string
 	AdapterPaths   []string
 	ProjectorPaths []string
-	Template       string
 	System         string
 	License        []string
 	Digest         string
 	Options        map[string]interface{}
 	Messages       []Message
+
+	Template *template.Template
```

Comment:
Do we need to make this a`*template.Template` (vs `string`)? Unless I missed one – I don't see a case where we end up calling `.String()` on it anyways elsewhere in this PR.

Reply:
`template.Template` encapsulates the model template and shifts templating responsibilities to that type from the server. this is used extensively in follow up changes where it takes ownership of template execution

passing around a concrete template type allows easy access to the template's capabilities (through `(*template.Template).Vars()`) which isn't used in this change but will be used in a follow up